@here/harp-map-controls/test/MapControlsTest.ts

```diff
@@ -530,4 +538,93 @@ describe("MapControls", function () {
             });
         }
     });
+
+    describe("toggletilt with inertia", () => {
+        beforeEach(function () {
+            // This tests runs a non mocked version of MapView, hence we need to mock some other
+            // methods to get it working correctl.
+            const clearColorStub: sinon.SinonStub = sandbox.stub();
+            sandbox
+                .stub(THREE, "WebGLRenderer")
+                .returns(TestUtils.getWebGLRendererStub(sandbox, clearColorStub));
+            sandbox
+                .stub(THREE, "WebGL1Renderer")
+                .returns(TestUtils.getWebGLRendererStub(sandbox, clearColorStub));
+            canvas = ({
+                clientWidth: 1,
+                clientHeight: 1,
+                addEventListener: sinon.stub(),
+                removeEventListener: sinon.stub()
+            } as unknown) as HTMLCanvasElement;
+            mapViewOptions = {
+                canvas,
+                // Both options cause the `addDataSource` method to be called, which we can't
+                // `await` on because it is called in the constructor, but we can disable them being
+                // added.
+                addBackgroundDatasource: false,
+                enablePolarDataSource: false
+            };
+            mapView = new MapView(mapViewOptions);
+            mapControls = new MapControls(mapView);
+        });
+
+        it("toggle tilt reaches configured tilt angle and 0", async () => {
+            mapControls.inertiaEnabled = true;
+            mapControls.toggleTilt();
+
+            let resolvePromise: (value?: unknown) => void;
+            let rejectPromise: (value?: unknown) => void;
+
+            {
+                let previousTilt: number | undefined;
+
+                // Tilt to "mapControls.tiltAngle"
+                const tiltCamera = new Promise((resolve, reject) => {
+                    resolvePromise = resolve;
+                    rejectPromise = reject;
+                });
+                const checkReachesTiltAngle = () => {
+                    const mapViewTiltRad = THREE.MathUtils.degToRad(mapView.tilt);
+
+                    // Resolve when we have reached within EPSILON of the tiltAngle
+                    if (Math.abs(mapViewTiltRad - mapControls.tiltAngle) <= Number.EPSILON) {
+                        mapView.removeEventListener(
+                            MapViewEventNames.AfterRender,
+                            checkReachesTiltAngle
+                        );
+                        resolvePromise();
+                    } else if (previousTilt === mapViewTiltRad) {
+                        rejectPromise();
+                    }
+                    previousTilt = mapViewTiltRad;
+                };
+                mapView.addEventListener(MapViewEventNames.AfterRender, checkReachesTiltAngle);
+                await tiltCamera;
```

Comment:
```suggestion
               expect(tiltCamera).to.eventually.be.closeTo(targetTilt, Number.EPSILON).then( // add here the second part of the test);
```
Using chai as promised the test expectation is clearer and the error messages will be more explicit. You need to import chai as promised (see TileLoaderTest).